### PR TITLE
Issue #2088 configure web and worker dynos separately

### PR DESCRIPTION
**Pull request by alecpm**
## Description
Adds `dyno_type_web` and `dyno_type_worker` configuration options which override the default `dyno_type`.

## Motivation and Context
See issue #2088

## How Has This Been Tested?
New automated test for heroku dyno scaling.

### CODE REVIEW

File: docs/source/configuration.rst [hunk 1]
```diff
@@ -258,14 +258,23 @@ very large difference to how the application behaves.
...
+    as the default for both web and worker dyno types. The minimum recommended is
+    ``standard-1x``, which should be sufficient for experiments that do not rely on
+    real-time coordination, such as :doc:`demos/bartlett1932/index`. Experiments that
+    require significant power to process websocket events should consider the higher
+    levels, ``standard-2x``, ``performance-m`` and ``performance-l``. In all but the
+    most intensive experiments, either ``dyno_type`` or ``num_dynos_web`` should be
+    increased, not both. See ``dyno_type_web`` and ``dyno_type_worker`` below
+    for information about more specific settings.
+
+``dyno_type_web``
```
Comment (jessesnyder) [inline docs/source/configuration.rst:271]: I'd spell out that this (and `dyno_type_worker`) will override the value of `dyno_type` if provided.

File: dallinger/deployment.py [hunk 2]
```diff
@@ -410,8 +410,9 @@ def deploy_sandbox_shared_setup(log, verbose=True, app=None, exp_config=None):
     git.push(remote="heroku", branch="HEAD:master")
 
     log("Scaling up the dynos...")
-    size = config.get("dyno_type")
+    default_size = config.get("dyno_type")
     for process in ["web", "worker"]:
+        size = config.get("dyno_type_" + process, default_size)
```
Comment (jessesnyder) [inline dallinger/deployment.py:415]: Well that worked out nicely!

File: tests/test_deployment.py [hunk 3]
```diff
@@ -457,6 +457,15 @@ def test_scales_dynos(self, dsss, heroku_mock, active_config):
             ]
         )
 
+    def test_scales_different_dynos(self, dsss, heroku_mock, active_config):
+        active_config.set("dyno_type", u"ignored")
+        active_config.set("dyno_type_web", u"tiny")
+        active_config.set("dyno_type_worker", u"massive")
+        dsss(log=mock.Mock())
+        heroku_mock.scale_up_dyno.assert_has_calls(
+            [mock.call("web", 1, u"tiny"), mock.call("worker", 1, u"massive"),]
```
Comment (jessesnyder) [inline tests/test_deployment.py:466]: Looks like `black` set you up for a scolding by `flake8`(?)

Comment (alecpm) [reply tests/test_deployment.py:466]: Ugh, that's exactly what happened, I wrote it flake8 compatible and then black put it on a single line.

File: docs/source/configuration.rst (hunk 1 above)
Comment (alecpm) [reply docs/source/configuration.rst:271]: Done.

**Pull request merged**